### PR TITLE
cleanup functions of Hermitian matrices

### PR DESCRIPTION
**Pull request by araujoms**
The functions of Hermitian matrices are a bit of a mess. For example, if we have a Hermitian matrix `a` with negative eigenvalues, `a^0.5` doesn't produce the `Symmetric` wrapper, but `sqrt(a)` does. On the other hand, if we have a positive definite `b`, `b^0.5` will be `Hermitian`, but `sqrt(b)` will be `Symmetric`:
```julia
using LinearAlgebra
a = Hermitian([1.0 2.0;2.0 1.0])
a^0.5
sqrt(a)
b = Hermitian([2.0 1.0; 1.0 2.0])
b^0.5
sqrt(b)
```
This sort of arbitrary assignment of wrappers happens with pretty much all functions defined there. There's also some oddities, such as `cis` being the only function defined for `SymTridiagonal`, even though all `eigen`-based functions work, and `cbrt` being the only function not defined for complex Hermitian matrices.

I did a cleanup: I defined all functions for `SymTridiagonal` and `Hermitian{<:Complex}`, and always assigned the appropriate wrapper, preserving the input one when possible.

There's an inconsistency remaining that I didn't fix, that only `sqrt` and `log` accept a tolerance argument, as changing that is probably breaking.

There were also hardly any tests that I could find (only `exp`, `log`, `cis`, and `sqrt`). I'm happy to add them if it's desired.

### CODE REVIEW

File: stdlib/LinearAlgebra/src/symmetric.jl [hunk 1]
```diff
@@ -855,16 +861,25 @@ function ^(A::Hermitian{T}, p::Real) where T
             return Hermitian(retmat)
         end
     else
-        return (F.vectors * Diagonal((complex.(F.values).^p))) * F.vectors'
+        retmat = (F.vectors * Diagonal((complex.(F.values).^p))) * F.vectors'
+        if T <: Real
+            return Symmetric(retmat)
```
Comment (jishnub) [inline stdlib/LinearAlgebra/src/symmetric.jl:866]: Can this be `Hermitian(retmat)` instead? This should narrow the possible return types.

Comment (jishnub) [reply stdlib/LinearAlgebra/src/symmetric.jl:866]: Oh, I guess `retmat` is complex in any case, so this won't apply. In that case, does returning a `Symmetric` complex matrix provide an advantage over a `Matrix`?

Comment (araujoms) [reply stdlib/LinearAlgebra/src/symmetric.jl:866]: Indeed, `retmat` is not Hermitian. Complex symmetric matrices are of limited utility, I mainly did it to be consistent with the `sqrt` function, which was already applying the wrapper.

Comment (araujoms) [reply stdlib/LinearAlgebra/src/symmetric.jl:866]: I looked it up, and it turns out one solve linear systems with complex symmetric matrices via `sysv!`. The function does not accept the wrapper, though, it wants a raw matrix as input.

**Review by jishnub — APPROVED** [review]
Looks good to me overall

**Pull request merged**